qiskit_experiments/library/randomized_benchmarking/rb_experiment.py

```diff
@@ -132,89 +133,335 @@ def circuits(self) -> List[QuantumCircuit]:
 
         Returns:
             A list of :class:`QuantumCircuit`.
+
+        Raises:
+            QiskitError: if basis_gates is not set in transpile_options nor in backend configuration.
         """
         rng = default_rng(seed=self.experiment_options.seed)
         circuits = []
+        if not hasattr(self.transpile_options, "basis_gates"):
+            raise QiskitError("transpile_options.basis_gates must be set for rb_experiment")
+
+        self.load_transpiled_cliff_circuits()
         for _ in range(self.experiment_options.num_samples):
-            circuits += self._sample_circuits(self.experiment_options.lengths, rng)
+            rb_circuits, _ = self._build_rb_circuits(self.experiment_options.lengths, rng)
+            circuits += rb_circuits
         return circuits
 
-    def _sample_circuits(self, lengths: Iterable[int], rng: Generator) -> List[QuantumCircuit]:
-        """Return a list RB circuits for the given lengths.
+    def load_transpiled_cliff_circuits(self):
+        """
+        Load the transpiled clifford circuits from the file into an array of
+        quantum circuits
 
-        Args:
-            lengths: A list of RB sequences lengths.
-            seed: Seed or generator object for random number
-                  generation. If None default_rng will be used.
+        Raises:
+            QiskitError: if the file with transpiled cliffords does not exist.
 
-        Returns:
-            A list of :class:`QuantumCircuit`.
         """
-        circuits = []
-        for length in lengths if self._full_sampling else [lengths[-1]]:
-            elements = self._clifford_utils.random_clifford_circuits(self.num_qubits, length, rng)
-            element_lengths = [len(elements)] if self._full_sampling else lengths
-            circuits += self._generate_circuit(elements, element_lengths)
-        return circuits
+        transpiled_circs_file = CliffordUtils.file_name(
+            self.num_qubits, self.transpile_options.basis_gates
+        )
+        n = self.num_qubits
+        if self._transpiled_cliff_circuits[n] is None:
+            if os.path.isfile(transpiled_circs_file):
```

Comment:
This is no longer our standard experiment workflow. This seems like a research code, i.e. framework is performant but dedicated to very limited use case. We should be able to create circuit without caching. For example, we may run this experiment with multiple backends with different basis set.